test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -277,185 +298,202 @@ def _process_web_message(self, timeout):
+        if node_obm == []:
+            logs.trl.debug('Picked node does not have OBM settings. Trying to add them')
+            applied = self.__apply_obmsetting_to_node(self.__get_run_context('start_nodeid'))
+            self.assertTrue(applied, 'Failed to apply obm settings')
+
+    @depends(after='test_rediscover_check_obm')
+    def test_rediscover_reboot_kickoff(self):
+        nodeid = self.__get_run_context('start_nodeid')
+        # first setup the web-hook to monitor (see test_rediscover_rackd_discover_hook) for
+        # rackhd hook messages.
+        self.__set_web_hook()
+        # now give the node a kick
         response = fit_common.rackhdapi(
             '/redfish/v1/Systems/' + nodeid + '/Actions/ComputerSystem.Reset', action='post',
             payload={"reset_type": "ForceRestart"})
         self.assertTrue(
             response['status'] < 209, 'Incorrect HTTP return code, expected<209, got:' + str(response['status']))
         graphid = response['json']["@odata.id"].split('/redfish/v1/TaskService/Tasks/')[1]
-
-        # wait for workflow started message.
-        self._wait_amqp_message(10)
-        workflow_amqp = amqp_queue.get()
-        if workflow_amqp[0][0:14] == "started":
-            self._process_message("started", graphid, nodeid, "graph", workflow_amqp)
-        else:
-            self._process_message("progress.updated", graphid, nodeid, "graph", workflow_amqp)
-
-        # wait for progress update finish message.
-        self._wait_amqp_message(10)
-        workflow_amqp = amqp_queue.get()
-        self._process_message("progress.updated", graphid, nodeid, "graph", workflow_amqp)
-
-        # wait for progress finish message.
-        retry_count = 0
-        while retry_count < 10:
-            self._wait_amqp_message(60)
-            workflow_amqp = amqp_queue.get()
-            if workflow_amqp[0][0:14] == "graph.finished":
-                self._process_message("finished", graphid, nodeid, "graph", workflow_amqp)
-                break
-            retry_count = retry_count + 1
-        self.assertNotEquals(retry_count, 10, "No AMQP workflow finished message received")
-        reset_worker.dispose()
-
-    def _node_delete(self, nodeid):
-        logs.debug('launch node delete AMQP thread')
-        td = AmqpWorker(
-            exchange_name="on.events", topic_routing_key="node.#.information." + nodeid + ".#",
-            external_callback=self.amqp_callback, timeout=30)
-        td.setDaemon(True)
-        td.start()
+        self.__set_run_context('reboot_graphid', graphid)
+
+    @depends(after='test_rediscover_reboot_kickoff', before='test_rediscover_node_delete')
+    def test_rediscover_reboot_graph_ampq_flow(self):
+        nodeid = self.__get_run_context('start_nodeid')
+        graphid = self.__get_run_context('reboot_graphid')
+        self.__qproc.push_group(ordered=True)
+        # todo ^^^^ erika to help name that method!

```

Comment:
We can also push another unordered group (at least in theory!). The comment may have to do, since I'm coming up with nothing for how to name this right :(